api/src/main/java/com/microsoft/gctoolkit/GCToolKit.java

```diff
@@ -97,19 +105,21 @@ public void registerAggregation(Class<? extends Aggregation> aggregationClass) {
      * that were {@link #registerAggregation(Class) registered}, if appropriate for
      * the GC log file.
      *
-     * @param dataSource The log to analyze, typically a
+     * @param logFile The log to analyze, typically a
      *                   {@link SingleGCLogFile} or
      *                   {@link RotatingGCLogFile}.
      * @return a representation of the state of the Java Virtual Machine resulting
      * from the analysis of the GC log file.
      */
-    public JavaVirtualMachine analyze(DataSource<?> dataSource) {
+    public JavaVirtualMachine analyze(GCLogFile logFile) throws IOException {
+        //todo: revert this to DataSource to account for non-GC log data sources once we have a use case (maybe JFR but JFR timers drift badly ATM)
         // Potential NPE, but would have logged if there was trouble creating the instance.
-        JavaVirtualMachine javaVirtualMachine = loadJavaVirtualMachine();
+        JavaVirtualMachine javaVirtualMachine = loadJavaVirtualMachine(logFile);
+        //todo: do we need reflection????
```

Comment:
You'd still need to load the impl class using the classloader since it's implemented down in vertx, which api doesn't know about. Also, this method returns a JavaVirtualMachine, so you end up need to construct it anyway (alternatively, the 'analyze' method that is invoked could return JavaVirtualMachine). If performance is an issue, then MH is the way to go. But here performance isn't a concern since the reflection overhead is overshadowed by the parsing itself.

Reply:
I think this can all be collapsed so that we won't need impl classes. Reflection overhead isn't a concern, it's more of, the target will be a direct method call.